app/main/views/agreements.py

```diff
@@ -237,3 +237,56 @@ def put_signed_framework_agreement_on_hold(agreement_id):
         return jsonify(message="Database Error: {0}".format(e)), 400
 
     return jsonify(agreement=framework_agreement.serialize())
+
+
+@main.route('/agreements/<int:agreement_id>/countersign', methods=['POST'])
+def countersign_agreement(agreement_id):
```

Comment:
Part of me is starting to prefer the term `approve` rather than `countersign` for this action, as it records an admin user approving the countersignature, rather than the countersigning itself.  Maybe one for discussion, or maybe we just stick with it how it is rather than introducing yet another term.

But maybe adding `approved` as an additional status would clear up the logic in the frontend that checks for a countersigned file path rather than for the status of the FA.  If we added `approved` as a status we could have `approved` = good to go but no countersigned file exists yet and `countersigned` = file has been generated and made available to the supplier.